graphql_api/types/bundle_analysis/base.py

```diff
@@ -135,6 +145,65 @@ def resolve_assets(
     return list(bundle_report.assets())
 
 
+@bundle_report_bindable.field("assetsPaginated")
+def resolve_assets_paginated(
+    bundle_report: BundleReport,
+    info: GraphQLResolveInfo,
+    ordering: AssetOrdering = AssetOrdering.SIZE,
+    ordering_direction: OrderingDirection = OrderingDirection.DESC,
+    first: Optional[int] = None,
+    after: Optional[str] = None,
+    last: Optional[int] = None,
+    before: Optional[str] = None,
+) -> Dict[str, object]:
+    # All filtered assets before pagination
+    assets = list(
```

Comment:
A non-ideal way to perform pagination. It is still fetching all the assets (minus the ones filtered out) from the report and then return the requested blocks. Since fetching all assets is trivial for a given report we will live with this non optimized way. The major reason we want to paginate assets is because each asset object contains timeseries measurement data which is what's causing 99% of the load time. Paginating the assets mean we can lazy load the timeseries data on the UI. Note that fetching all assets in the first step doesn't begin the computation of the measurements data which means its fine.